src/synthesizer/golang/mod.rs

```diff
@@ -620,25 +620,37 @@ impl GolangEmitter for ResourceIr {
                 }
             }
             Self::Object(structure, properties) => {
+                let mut structure_is_simple_json = false;
                 let props = output.indent_with_options(IndentOptions {
                     indent: INDENT,
                     leading: Some(match structure {
                         Structure::Composite(name) => match *name {
                             "Tag" => "&cdk.CfnTag{".into(),
                             name => format!("&{name}/* FIXME */{{").into(),
                         },
-                        Structure::Simple(cfn) => {
-                            unreachable!("object with simple structure ({:?})", cfn)
+                        Structure::Simple(cfn) => match cfn {
+                            CfnType::Json => {
+                                structure_is_simple_json = true;
+                                "map[string]interface{} {".into()
+                            }
+                            _ => unreachable!("object with simple structure ({:?})", cfn)
```

Comment:
Now this case should be truly unreachable, because there should never be an `Object` where its structure is one of the basic types.